R/dfm.R

```diff
@@ -436,14 +447,15 @@ compile_dfm.tokens <- function(x, verbose = TRUE) {
     x <- unclass(x)
     
     # shift index for padding, if any
-    index <- unlist(x, use.names = FALSE)
+    features_index <- unlist(x, use.names = FALSE)
     if (attr(x, 'padding')) {
         types <- c("", types)
-        index <- index + 1
+        features_index <- features_index + 1
     }
+    documents_index <- rep(seq_len(length(x)), lengths(x))
     
-    temp <- sparseMatrix(j = index, 
-                         p = cumsum(c(1, lengths(x))) - 1, 
+    temp <- sparseMatrix(i = documents_index,
```

Comment:
I am thinking to do document grouping by re-indexing document ids, but you can keep current version.